### ISSUE TITLE
Investigate and improve CYS assembler hub performance

### ISSUE THREAD

**Issue by chihsuan** [woothemes/woocommerce#40366]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Related: https://github.com/woocommerce/team-ghidorah/issues/265

Deep dive into CYS assembler hub and make some improvements

**Comment by rjchow** [comment]
After some investigation these were the findings:
- Having devtools open severely degrades the performance, as every iframe loads the site's JS and CSS in its entirety. Every color, font, header, footer, template variation block is an iframe, and since if caching is disabled by devtools, all of these assets have to be loaded again for each one. Caching mitigates this significantly
- Using the a8c proxy introduces additional round trip latency and bandwidth impacts, compounding on the above
- Font loading could be optimised, and https://github.com/woocommerce/woocommerce/pull/40458 does that
- There is some memory leak or zombie computation in the background that causes the entire browser's performance to degrade with the site assembler active. This should be investigated at some point but it is not immediately urgent as it should be uncommon for end users to have the site assembler open for hours like we do

TL;DR: shut your devtools and don't use a8c proxy when giving demos
